pkg/oauth/tokenrequest/request_token.go

```diff
@@ -63,6 +67,16 @@ func RequestToken(clientCfg *restclient.Config, challengeHandlers ...challengeha
 	return NewRequestTokenOptions(clientCfg, false, challengeHandlers...).RequestToken()
 }
 
+// RequestTokenWithAuthorization uses the OAuth Authorization Code Grant flow in order to obtain an authorization code
+// and exchange it for an access token. It creates and starts a local HTTP server on the loopback address and the
+// specified port, and it invokes the specified authorization handler func so that the user can define how the
+// authorization URL should be handled (e.g. opened in a browser). The URL uses a suitable OAuth client with a
+// redirect to the local HTTP server which contains the authorization request that can be exchanged for an
+// access token.
+func RequestTokenWithAuthorization(clientCfg *restclient.Config, authzUrlHandler func(string) error, callbackPort int) (string, error) {
+	return NewRequestTokenOptions(clientCfg, false).RequestTokenWithAuthorization(authzUrlHandler, callbackPort)
```

Comment:
```go
// comment here
type AuthorizationURLHandlerFunc func(url *url.URL) error

// comment here
func RequestTokenWithLocalCallback(clientCfg *restclient.Config, handleAuthzURL AuthorizationURLHandlerFunc, localCallbackPort int) {
    return NewRequestTokenOptions(clientCfg, false).WithLocalCallback(handleAuthzURL, localCallbackPort).RequestToken()
}
```

`WithLocalCallback()` should add an unstarted local callback server to the options, among with the authz URL handler

Reply:
Nice idea; fixed here: https://github.com/openshift/library-go/pull/1531/files#diff-f30259f7a13841a226555f2801678f096da8bb80456cf414201b8cfd647da728R124-R133